app/models/mentor/discussion.rb

```diff
@@ -111,13 +110,4 @@ def student_handle = anonymous_mode? ? "anonymous" : student.handle
   def student_flair = anonymous_mode? ? "anonymous" : student.flair
   def student_name = anonymous_mode? ? "User in Anonymous mode" : student.name
   def student_avatar_url = anonymous_mode? ? nil : student.avatar_url
-
-  def update_stats!
-    Mentor::UpdateStats.defer(
-      mentor,
-      update_counts: previous_changes.key?('status'),
-      update_satisfaction_rating: previous_changes.key?('rating')
-    )
-    Mentor::Discussion::UpdateNumFinishedDiscussions.defer(self) if previous_changes.key?('status') && finished?
-  end
```

Comment:
We now only call all this during the finishing up process of a solution. Which makes much more sense!